chains/nomad-ethereum/src/gelato/mod.rs

```diff
@@ -0,0 +1,218 @@
+use ethers::signers::Signer;
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::{Address, H256};
+use ethers::{prelude::Bytes, providers::Middleware};
+use gelato_relay::{GelatoClient, RelayResponse, TaskState};
+use nomad_core::{ChainCommunicationError, Signers, TxOutcome};
+use std::{str::FromStr, sync::Arc};
+use tokio::task::JoinHandle;
+use tokio::time::{sleep, Duration};
+use tracing::info;
+use utils::CHAIN_ID_TO_FORWARDER;
+
+/// EIP-712 forward request structure
+mod types;
+pub use types::*;
+
+pub mod utils;
+
+pub(crate) const ACCEPTABLE_STATES: [TaskState; 4] = [
```

Comment:
Why not 

```
impl TaskState {
    fn <descriptive_name>(&self) -> bool {
       ...
    }
}
```

Reply:
Would rather not implement any extra logic in `gelato-relay` crate (where `TaskState` lives). `rust/gelato-relay` should have no knowledge of what an acceptable state is for nomad agents sending requests. Similar reasoning as https://github.com/nomad-xyz/rust/pull/178#discussion_r885098350